### PR TITLE
Add creating new message tutorial

### PR DESCRIPTION
**Pull request by pablogs9**
This PR adds a tutorial about creating a custom message type in micro-ROS

### CODE REVIEW

File: _docs/tutorials/core/create_new_type/index.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+cd firmware/mcu_ws
+ros2 pkg create --build-type ament_cmake my_custom_message
+cd my_custom_message
+mkdir msg
+touch msg/MyCustomMessage.msg
+```
+
+In the autogenerated `CMakeLists.txt` file you should add the following lines just before `ament_package()`:
+
+```
```
Comment (jamoralp) [inline _docs/tutorials/core/create_new_type/index.md:20]:
```suggestion
```cmake
```

File: _docs/tutorials/core/create_new_type/index.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+rosidl_generate_interfaces(${PROJECT_NAME}
+  "msg/MyCustomMessage.msg"
+ )
+...
+```
+
+
+In the autogenerated `package.xml` file you should add the following lines:
+
+```
```
Comment (jamoralp) [inline _docs/tutorials/core/create_new_type/index.md:33]:
```suggestion
```xml
```

File: _docs/tutorials/core/create_new_type/index.md [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+uint8 uint8_test
+int16 int16_test
+uint16 uint16_test
+int32 int32_test
+uint32 uint32_test
+int64 int64_test
+uint64 uint64_test
+```
+
+Now, you can build your micro-ROS workspace as usual. Check [**First micro-ROS application on an RTOS**](../first_application_rtos/) for more details of how to build a micro-ROS environment.
```
Comment (jamoralp) [inline _docs/tutorials/core/create_new_type/index.md:61]:
```suggestion
Now, you can build your micro-ROS workspace as usual. Check [**First micro-ROS application on an RTOS**](../first_application_rtos/) for more details on how to build a micro-ROS environment. 
```

File: _docs/tutorials/core/create_new_type/index.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
+---
+title: How to create a new micro-ROS type
+permalink: /docs/tutorials/core/create_new_type/
+---
+
+This tutorial starts in an already create micro-ROS environment. Check [**First micro-ROS application on an RTOS**](../first_application_rtos/) for more details of how to build a micro-ROS environment.
```
Comment (FranFin) [inline _docs/tutorials/core/create_new_type/index.md:6]:
```suggestion
This tutorial starts in a previously created micro-ROS environment. Check [**First micro-ROS application on an RTOS**](../first_application_rtos/) for more details on how to build a micro-ROS environment.
```

File: _docs/tutorials/core/create_new_type/index.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+uint8 uint8_test
+int16 int16_test
+uint16 uint16_test
+int32 int32_test
+uint32 uint32_test
+int64 int64_test
+uint64 uint64_test
+```
+
+Now, you can build your micro-ROS workspace as usual. Check [**First micro-ROS application on an RTOS**](../first_application_rtos/) for more details on how to build a micro-ROS environment.
```
Comment (FranFin) [inline _docs/tutorials/core/create_new_type/index.md:59]: Not sure what you mean here. You already said this at the beginning of the tutorial.

File: _docs/tutorials/core/create_new_type/index.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+rcl_publish(&publisher, &msg, NULL);
+ 
+...
+```
+
+You can find further information in the [ROS 2 Create custom ROS 2 msg and srv files](https://index.ros.org/doc/ros2/Tutorials/Custom-ROS2-Interfaces).
+
+## Using type composition
+
+It is possible to create custom types that include members from another ROS 2 message types packages. For example let's add a memeber with type `Point32` from the ROS 2 package `geometry_msgs`.
```
Comment (FranFin) [inline _docs/tutorials/core/create_new_type/index.md:85]:
```suggestion
It is possible to create custom types that include members from another ROS 2 message types packages. For example let's add a member with type `Point32` from the ROS 2 package `geometry_msgs`.
```

File: _docs/tutorials/core/create_new_type/index.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+```xml
+...
+<build_depend>rosidl_default_generators</build_depend>
+<exec_depend>rosidl_default_runtime</exec_depend>
+<member_of_group>rosidl_interface_packages</member_of_group>
+<depend>geometry_msgs</depend>
+...
+```
+
+The message definition in `msg/MyCustomMessage.msg` now can add types from `geometry_msgs` package:
```
Comment (FranFin) [inline _docs/tutorials/core/create_new_type/index.md:111]:
```suggestion
The message definition in `msg/MyCustomMessage.msg` can now include types from the `geometry_msgs` package:
```

File: _docs/tutorials/core/create_new_type/index.md [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+
+```
+...
+int64 int64_test
+uint64 uint64_test
+geometry_msgs/Point32 point32_test
+```
+
+And finally, in your code you can access to this new member of your custom type:
```
Comment (FranFin) [inline _docs/tutorials/core/create_new_type/index.md:120]:
```suggestion
And finally, in your code you can access this new member of your custom type:
```

File: _docs/tutorials/core/create_new_type/index.md (hunk 5 above)
Comment (pablogs9) [reply _docs/tutorials/core/create_new_type/index.md]: Please check changes

File: _docs/tutorials/core/create_new_type/index.md [hunk 9]
```diff
@@ -0,0 +1,138 @@
...
+uint8 uint8_test
+int16 int16_test
+uint16 uint16_test
+int32 int32_test
+uint32 uint32_test
+int64 int64_test
+uint64 uint64_test
+```
+
+Now, you can build your micro-ROS workspace as usual. As explained in [**First micro-ROS application on an RTOS**](../first_application_rtos/) the `ros2 run micro_ros_setup build_firmware.sh` command will build all packages located inside `mcu_ws`.
```
Comment (FranFin) [inline _docs/tutorials/core/create_new_type/index.md:59]:
```suggestion
Now, you can build your micro-ROS workspace as usual. As explained in [**First micro-ROS application on an RTOS**](../first_application_rtos/), the `ros2 run micro_ros_setup build_firmware.sh` command will build all packages located inside `mcu_ws`. 
```

**Pull request merged**